### PR TITLE
Fix issue #369

### PR DESCRIPTION
**Pull request by Sobak**
Code provided by @raimaj, which accordingly to my tests should solve issue #369. I didn't looked into inner workings very deeply, so let me know if you're concerned of the implementation so that we can figure out something better.

Thanks

### CODE REVIEW

File: src/PanelTraits/Create.php [hunk 1]
```diff
@@ -112,10 +112,12 @@ public function syncPivot($model, $data, $form = 'create')
...
                 $values = isset($data[$field['name']]) ? $data[$field['name']] : [];
-                if ($model->{$field['name']}) {
-                    $model->{$field['name']}()->update($values);
-                } else {
-                    $model->{$field['name']}()->create($values);
+
+                foreach ($values as $value) {
+                    $class = $field['model'];
+                    $class = new $class();
+                    $ref = $class::find($value);
```
Comment (lloy0076) [inline src/PanelTraits/Create.php]: What **is** `$class` here:

a) A string?
b) An instantiated object?

It looks like it's an instantiated object here ... and although PHP (being icky) probably allows static calls on instantiated objects it's a bit ick imho.

Comment (Sobak) [inline src/PanelTraits/Create.php]: You're right, fixed

**Comment by Sobak** [comment]
Fixed to use dynamic call

**Comment by Sobak** [comment]
@tabacitu Hi! Any hope for merging it or any issues you see? Having it fixed in the application involves some overriding and is not a cleanest solution...

File: src/PanelTraits/Create.php [hunk 2]
```diff
@@ -112,10 +112,12 @@ public function syncPivot($model, $data, $form = 'create')
...
                 $values = isset($data[$field['name']]) ? $data[$field['name']] : [];
-                if ($model->{$field['name']}) {
-                    $model->{$field['name']}()->update($values);
-                } else {
-                    $model->{$field['name']}()->create($values);
+
+                foreach ($values as $value) {
+                    $class = $field['model'];
+                    $class = new $class();
+                    $ref = $class->find($value);
```
Comment (lloy0076) [inline src/PanelTraits/Create.php]: What if that turns out to be `null`? i.e. for some reason `$value` isn't there?

**Comment by gotrecillo** [comment]
@Sobak Hi, could you explain what you did in your application to fix until it is merged in master ?

File: src/PanelTraits/Create.php (hunk 2 above)
Comment (gotrecillo) [reply src/PanelTraits/Create.php]: check this https://github.com/gotrecillo/CRUD/commit/ea927d3f2b2325b932a80c4bed2633b784519591

Solves the ``` $value = null``` you described + allows for updating

**Comment by Sobak** [comment]
@lloy0076 @gotrecillo @tabacitu ok, so the previous approach was wrong, on each save of the entitity it caused records in the pivot table to duplicate, which was obviously not the expected behavior. So I sat down and replaced it with a code which not only works but it's also more straightforward.

Basically it does exactly what the code shown in the fork linked above - skips rows that already exists - and this is functionallity provided by Laravel's `sync()` method out-of-the-box. This solution is tested and running fine for a last month or so.

Can we please merge it and release the fixed version? It's quite a shame that one of the functionallities described in the docs doesn't work for at least 3 months, throwing fatal errors and noone seems to really care :/

**Comment by tabacitu** [comment]
Hi @Sobak ,

Thanks a lot for the PR - I totally understand your frustration, handling all incoming requests for Backpack has been a huge endeavour in the past few months, but generous people like @lloy0076 and yourself have been of great help.

I've merged your PR into ```dev``` and it will be available as a patch update shortly. So if you do a ```composer update``` tomorrow you'll have the new & improved syncPivot.

Thanks!

**Comment by Sobak** [comment]
@tabacitu thanks a lot! Indeed I got frustrated but it's good to see that you're responsive. Expect some more PRs probably :)

**Pull request merged**